alt_src/alt_src.py

```diff
@@ -70,6 +70,8 @@
 CONFIG_INT_OPTS = set([])
 CONFIG_BOOL_OPTS = set(['smtp_enabled', 'push_tags', 'debrand'])
 
+logging.basicConfig(filename=CONFIG_DEFAULTS["log_file"])
```

Comment:
Two issues with this:

- this shouldn't be done at the top level like this, because it means even just importing the file is going to change loggers (as is done e.g. by tests).  If we'll call logging.basicConfig, then can you please have that inside of "main" somewhere?

- alt-src already has a `setup_logging` function and I'm not sure these are going to work together correctly.  As I understand it, this logging.basicConfig is going to install a default handler, but then in setup_logging, another StreamHandler to sys.stdout is added.

Due to the second problem, I would suspect that this could result in duplicate log messages, i.e. some logs coming out twice instead of once.  I haven't tested it though.  Have you tried it?

Reply:
@rohanpm, the trouble is the handlers of a logger are set up after `options.config = get_config(options.cfile, options.copts)`,
but the `get_config` process can also be corrupted and need a logger to track this.
So if the something wrong happens in `get_config`, then there is no handler.

Reply:
Yeah, I understand that.

In another comment I've suggested changes to setup_logging which could make this work correctly.  As well as that, a couple suggestions here:

- don't pass "filename".  What's the point?  `CONFIG_DEFAULTS["log_file"]` is always `None`.
- I think a comment explaining this would be worthwhile.  Like "We don't know the desired log config until we've read the config file, but we might need to log something before then - so ensure there's at least some basic config (to be replaced later)."